rln/rln.go

```diff
@@ -393,6 +394,62 @@ func (r *RLN) GetLeaf(index MembershipIndex) (IDCommitment, error) {
 	return result, nil
 }
 
+// GetMerkleProof returns the Merkle proof for the element at the specified index
+// The output should be parsed as: num_elements<8>|path_elements<var1>|num_indexes<8>|path_indexes<var2>
+// where num_elements indicate var1 array size and num_indexes indicate var2 array size.
+// Both num_elements and num_indexes shall be equal and match the tree depth.
+// A tree with depth 20 returns 676 bytes = 8 + 32 * 20 + 8 + 20 * 1
+// Proof elements are stored as little endian
+func (r *RLN) GetMerkleProof(index MembershipIndex) (MerkleProof, error) {
+	proofBytes, err := r.w.GetMerkleProof(index)
+	if err != nil {
+		return MerkleProof{}, err
+	}
+
+	var result MerkleProof
+	var numElements big.Int
+	var numIndexes big.Int
+
+	offset := 0
+
+	// Get amounf of elements in the proof
+	numElements.SetBytes(revert(proofBytes[offset : offset+8]))
+	offset += 8
+
+	result.PathElements = make([]MerkleNode, numElements.Uint64())
+
+	for i := uint64(0); i < numElements.Uint64(); i++ {
+		copy(result.PathElements[i][:], proofBytes[offset:offset+32])
+		offset += 32
+	}
+
+	// Get amount of indexes in the path
+	numIndexes.SetBytes(revert(proofBytes[offset : offset+8]))
```

Comment:
Also probably makes sense to check whether there are enough bytes in the array left!

Reply:
same